l10n_ch_delivery_carrier_label_quickpac/quickpac/web_service.py

```diff
@@ -0,0 +1,517 @@
+            franking_license=company.quickpac_franking_license,
+            customer=customer,
+        )
+        return file_infos
+
+    def _prepare_recipient(self, picking):
+        """Create a Recipient for a partner from a picking
+
+        :param picking: a picking record
+        :return: Recipient
+        """
+        partner = self._get_recipient_partner(picking)
+        partner_name = partner.name or partner.parent_id.name
+        email = parseaddr(partner.email)[1]
+        recipient = Recipient(
+            name1=sanitize_string(partner.name),
+            street=sanitize_string(partner.street),
+            zip=partner.zip,
+            city=partner.city,
+            e_mail=email or None,
+        )
+
+        if partner.country_id.code:
+            recipient.country = partner.country_id.code.upper()
+        if partner.street2:
+            recipient.address_suffix = sanitize_string(partner.street2)
+
+        parent_id = partner.parent_id
+        if parent_id and parent_id.name != partner_name:
+            recipient.name2 = sanitize_string(parent_id.name)
+            recipient.personally_addressed = False
+
+        # Phone and / or mobile should only be displayed if instruction to
+        # Notify delivery by telephone is set
+        is_phone_required = [
+            option for option in picking.option_ids if option.code == "ZAW3213"
+        ]
+        if is_phone_required:
+            phone = partner.phone or parent_id and parent_id.phone
+            if phone:
+                recipient.phone = phone
+
+            mobile = partner.mobile or parent_id and parent_id.mobile
+            if mobile:
+                recipient.mobile = mobile
+
+        return recipient
+
+    def _prepare_attributes(self, picking, pack_counter=None, pack_total=None):
+        """Define specific attributes for a delivery
+
+        :param picking: a picking record
+        :param pack_counter: the current package index
+        :param pack_total: the total packages number
+        :return: ServiceCodeAttributes
+        """
+        services = [
+            code
+            for codes in (
+                option.code.split(",")
+                for option in picking.option_ids
+                if option.tmpl_option_id.quickpac_type
+                in ("basic", "additional", "delivery")
+            )
+            for code in codes
+        ]
+        if not services:
+            raise UserError(_("Missing required delivery option on picking."))
+
+        dimensions = Dimensions(weight=picking.shipping_weight)
+        attributes = ServiceCodeAttributes(przl=services, dimensions=dimensions)
+        return attributes
+
+    def _prepare_notification(self, picking):
+        """Define how and who will be notified
+
+        :param picking: a picking record
+        :return: Notification
+        """
+        communication = Communication(
+            email=picking.partner_id.email, mobile=picking.partner_id.mobile
```

Comment:
```suggestion
            email=picking.partner_id.email or None, mobile=picking.partner_id.mobile or None
```
In case the there is no phone or mobile the returned value must be null and not false